### PR TITLE
[Merged by Bors] - Derived `Copy` trait for `bevy_input` events, `Serialize`/`Deserialize` for events in `bevy_input` and `bevy_windows`, `PartialEq` for events in both, and `Eq` where possible in both.

### PR DESCRIPTION
**Pull request by targrub**
# Objective

Add traits to events in `bevy_input` and `bevy_windows`: `Copy`, `Serialize`/`Deserialize`, `PartialEq`, and `Eq`, as requested in https://github.com/bevyengine/bevy/issues/6022, https://github.com/bevyengine/bevy/issues/6023, https://github.com/bevyengine/bevy/issues/6024.

## Solution

Added the traits to events in `bevy_input` and `bevy_windows`.  Added dependency of `serde` in `Cargo.toml` of `bevy_input`.

## Migration Guide

If one has been `.clone()`'ing `bevy_input` events, Clippy will now complain about that.  Just remove `.clone()` to solve.

## Other Notes

Some events in `bevy_input` had `f32` fields, so `Eq` trait was not derived for them.
Some events in `bevy_windows` had `String` fields, so `Copy` trait was not derived for them.

### CODE REVIEW

File: crates/bevy_window/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ bevy_input = { path = "../bevy_input", version = "0.9.0-dev" }
 raw-window-handle = "0.4.2"
 
 # other
+serde = { version = "1.0", features = ["derive"] }
```
Comment (mockersf) [inline crates/bevy_window/Cargo.toml:23]: you've put use of serve behind a `serialize ` feature, but you didn't put this crate as optional and didn't add the `serialize` feature

Comment (targrub) [reply crates/bevy_window/Cargo.toml:23]: I did what I thought I saw `bevy_input` doing.  I'll change it to add those things.

Comment (targrub) [reply crates/bevy_window/Cargo.toml:23]: A little scary that the ci tests all passed.

Comment (mockersf) [reply crates/bevy_window/Cargo.toml]: You've added code that is always disabled and no tests on it. It's impossible for CI to detect that 😄

**Comment by mockersf** [comment]
now you need to add `bevy_window/serialize` to https://github.com/bevyengine/bevy/blob/28205fd3f4ef2dc2f7cb7cdb48511ebb34d97340/crates/bevy_internal/Cargo.toml#L48

File: crates/bevy_window/src/event.rs [hunk 2]
```diff
@@ -5,6 +5,7 @@ use bevy_math::{IVec2, Vec2};
 
 /// A window event that is sent whenever a window's logical size has changed.
 #[derive(Debug, Clone)]
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/event.rs:7]: PartialEq here for testing.

File: crates/bevy_window/src/event.rs [hunk 3]
```diff
@@ -15,6 +16,7 @@ pub struct WindowResized {
 
 /// An event that indicates that a new window should be created.
 #[derive(Debug, Clone)]
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/event.rs:18]:
```suggestion
#[derive(Debug, PartialEq, Clone)]
```

File: crates/bevy_window/src/event.rs (hunk 2 above)
Comment (alice-i-cecile) [reply crates/bevy_window/src/event.rs:7]:
```suggestion
#[derive(Debug, PartialEq, Clone)]
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
All the bevy_window events need PartialEq impls :) The general strategy LGTM now though.

File: crates/bevy_window/src/event.rs (hunk 2 above)
Comment (targrub) [reply crates/bevy_window/src/event.rs]: Got it.

File: crates/bevy_window/src/event.rs (hunk 3 above)
Comment (targrub) [reply crates/bevy_window/src/event.rs]: Got it.

File: crates/bevy_window/Cargo.toml (hunk 1 above)
Comment (targrub) [reply crates/bevy_window/Cargo.toml]: I understand.

**Review by alice-i-cecile — APPROVED** [review]
Awesome, LGTM now!

**Comment by targrub** [comment]
This PR's code should have ended with [29ebbd9](https://github.com/bevyengine/bevy/pull/6023/commits/29ebbd93e059d38eddc2de2f286465eff0ddae31)

File: crates/bevy_window/src/event.rs [hunk 4]
```diff
@@ -4,7 +4,8 @@ use super::{WindowDescriptor, WindowId};
 use bevy_math::{IVec2, Vec2};
 
 /// A window event that is sent whenever a window's logical size has changed.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```
Comment (bzm3r) [inline crates/bevy_window/src/event.rs:7]: Is it possible to derive `Copy` for the items in this file?

**Review by bzm3r — COMMENTED** [review]
Looks good to me! I just have one question.

**Comment by bzm3r** [comment]
Is it worth deriving `Hash` too, right now, or should that be left for a separate issue? For `bevy_input` items in particular, I can see that being useful.

File: crates/bevy_window/src/event.rs (hunk 4 above)
Comment (targrub) [reply crates/bevy_window/src/event.rs:7]: A number of them have `String` or `PathBuf` fields, so `Copy` cannot be derived for those.

**Comment by alice-i-cecile** [comment]
Flaky failure, see https://github.com/bevyengine/bevy/issues/6036

bors retry

**Pull request closed without merging**